call-home/src/main.rs

```diff
@@ -41,39 +50,97 @@ async fn main() -> anyhow::Result<()> {
+    let sleep_duration = match Duration::from_str(CALL_HOME_FREQUENCY) {
+        Ok(sleep_duration) => sleep_duration,
+        Err(err) => {
+            error!("failed to parse call-home frequency: {:?}", err);
+            std::process::exit(1);
+        }
+    };
 
-    let k8s_client = K8sClient::new().await?;
+    // Generate kubernetes client.
+    let k8s_client = match K8sClient::new().await {
+        Ok(k8s_client) => k8s_client,
+        Err(err) => {
+            error!("failed to generate Kubernetes client: {:?}", err);
+            std::process::exit(1);
+        }
+    };
 
-    let config = Configuration::new(endpoint, time::Duration::from_secs(30), None, None, true)
-        .map_err(|error| {
-            anyhow::anyhow!(
-                "Failed to create openapi configuration, Error: '{:?}'",
-                error
-            )
-        })?;
-    let client = openapi::clients::tower::ApiClient::new(config);
+    // Generate SHA256 hash of kube-system namespace UID.
+    let k8s_cluster_id = match k8s_client.get_cluster_id().await {
+        Ok(k8s_cluster_id) => digest(k8s_cluster_id),
+        Err(err) => {
+            error!("failed to get kubernetes cluster id: {:?}", err);
+            std::process::exit(1);
+        }
+    };
 
-    let mut report = generate_report(k8s_client.clone(), client.clone()).await;
-    report.deploy_namespace = namespace.clone();
-    report.product_version = version.clone();
+    // Generate receiver API client.
+    let receiver = match Receiver::new(&k8s_cluster_id) {
+        Ok(receiver) => receiver,
+        Err(err) => {
+            error!("failed to generate receiver API client: {:?}", err);
+            std::process::exit(1);
+        }
+    };
 
-    println!("{:?}", report);
+    // Generate Mayastor REST client.
+    let config = match Configuration::new(
+        endpoint,
+        time::Duration::from_secs(30),
+        None,
+        None,
+        true
+    ) {
+        Ok(config) => config,
+        Err(error) => {
+            error!("Failed to create openapi configuration, Error: '{:?}'", error);
+            std::process::exit(1);
+        },
+    };
+    let client = openapi::clients::tower::ApiClient::new(config);
 
     loop {
-        // TODO: For now it loops every 60 sec. Need to change this to 24hr and set the value in
-        // constants.
-        sleep(Duration::from_secs(60)).await;
-        let mut report = generate_report(k8s_client.clone(), client.clone()).await;
-        report.deploy_namespace = namespace.clone();
-        report.product_version = version.clone();
-        println!("{:?}", report);
+        // Generate report.
+        let report = generate_report(
+            k8s_client.clone(),
+            client.clone(),
+            k8s_cluster_id.clone(),
+            namespace.clone(),
+            version.clone(),
+        ).await;
+
+        // Encrypt data.
+        let output = match encrypt::encrypt(&report) {
```

Comment:
hmm I don't think it's a good idea to call encrypt directly, it does a lot of blocking calls which would hold up the async executor... have a look at tokio's `spawn_blocking`.